### DISCUSSION TITLE
Feeds from Reddit not working anymore

### DISCUSSION THREAD

**Discussion by roughnecks** [FreshRSS/FreshRSS#5944] [Q&A] [answered]
**Describe the bug**
Ops! Questo feed ha generato un errore. Per favore verifica se ancora disponibile.
Since a few days I'm getting this error for every subReddit Feed I follow.

**Expected behavior**
I'd expect to see the feeds even if the rss itself doesn't validate.

**Screenshots**
![image](https://github.com/FreshRSS/FreshRSS/assets/1666536/3d907287-dbcb-4bcf-8aeb-4e667c08ba73)

**Environment information (please complete the following information):**
- Device: Desktop PC
- OS: Windows 10
- Browser: Chrome
- FreshRSS version: 1.22.1
- Database version: sqlite3 3.40.1-2
- PHP version: PHP 8.2
- Installation type: Classic install from tar.gz

**Additional context**
Running on Debian 12.4

**Comment by Alkarex** [comment]
Try to curl from your server:

```sh
curl -v 'https://www.reddit.com/r/FreeGameFindings/.rss'
```

The most likely is that your IP has been blocked (your server, or the IP range of your data centre). Where are you hosted?

If that is the case, you have two main options:
* Move your FreshRSS instance to another IP, for instance at home;
* Use a proxy for your (problematic) feeds, the proxy being hosted elsewhere.

**Comment by roughnecks** [comment]
My VPS is hosted in Germany, by Contabo
I used to run the same setup on OVH, always in Germany, without issues.

Tried the curl command and in fact it says I'm blocked.

It says to make sure my user-agent is not empty and to contact them to discuss the blocking: is it worth doing that?
I don't have any other options so far.

Thanks

**Comment by roughnecks** [comment]
I contacted them - 2 times actually, because the first mail told me to use their web interface to send another request - and I explained the situation.

I also have an XMPP RSS Bot running with a Reddit feed, so it might be the issue as well.

Will let you know once I get news from reddit support.

**Comment by math-GH** [comment]
Strange, I guess Reddit is blocking IPs.... I use a VPN and get a message when opening https://www.reddit.com/r/FreeGameFinding/.rss:

![grafik](https://github.com/FreshRSS/FreshRSS/assets/1645099/71796b60-a3c0-43d1-9602-b79ce4ffbe54)

When VPN is disabled: I can open the feed.

So it is a IP address block. It does not help to set the user agent string.

**Comment by roughnecks** [comment]
yeah @math-GH , I got the same snippet from the curl command above.

Thanks for sharing!

**Comment by mustaruhtinas** [comment]
Not much that can be done, actually, I too am banned (error 403). Refresh interval 1h, no scraping or anything. Netcup VPS.
On home IP everything is fine.
I remember a few months ago there were lots of 429 errors. Seem they have become more aggressive now.
Edit: at least now I know there is no problem on my side.

**Comment by yuliu** [comment]
After some search, have got the following info that might help, or not:
 - https://forum.newsblur.com/t/redditt-feeds-no-longer-showing/10338/21
 - https://forum.newsblur.com/t/i-got-reddit-feeds-to-work-again/10365
 - https://www.reddit.com/r/redditdev/comments/18iuk10/403_when_accessing_rss_feeds/

Don't know if it's possible to use HTTP authentication for Reddit RSS retrieving?

**Comment by roughnecks** [comment]
yeah, thank you all, especially @yuliu for those links.

I tried changing a feed address to old.whatever but haven't got hopes up, since my IP is banned. I also have contacted their support but the first answer was this:

```
Thanks for writing in! We’re sorry you’re experiencing trouble with the site, but we’ll look into it
and reach out if we need any more information. In the meantime, we recommend logging in to the site or ensuring your script is connected via oAuth to regain access.
```

Will close this issue now, since it's not a FreshRSS issue.

**Comment by roughnecks** [comment, marked as answer]
Have to retake my words! Using old.reddit still works!

**Reply by Alkarex** [reply]
Good tip 👍🏻

**Comment by Mte90** [comment]
I have the same issue, in the log there is the alert but you have to search there cURL error 22: The requested URL returned error: 403 Blocked [https://www.reddit.com/r/commandline/top.rss?t=week&amp;limit=25] 

I confirm that with old.reddit works
